### ISSUE TITLE
Monster status tracking

### ISSUE THREAD

**Issue by Alphy11** [johreh/gloomycompanion#36]
Hey all,
I am embarrassed to say this seemed the best way to contact the contributors here  >.<
This is less of an issue, more of an open question.  Do you guys have any plans to expand this into more of a game-running assistant?  I am currently working on a project that allows multiple users to enter a session on their phones and from there they can keep track of initiative and monster HP.  The cool part is each player can edit hp from their own phones, and everyone else will be able to see the change in real time.  My playgroup has been having a lot of issues with this aspect of the game, so I was trying to automate it for us.  

I am wondering whether you guys are planning on doing any of this, to make sure we are not duplicating efforts :-)

**Comment by GinoGalotti** [comment]
Wow... phone support (and creating an API/Server side) has not been considered yet. We are (slowly) working on a way to show the monster stats, and when we have that we'd like to add a menu with all the active monsters, and a way to "punch" and do damage on them.

Initiative.. well, we are showing the cards all in one place. We're missing the initiative of the characters, but it shouldn't be hard to add an UI showing the initiative of all the cards. I don't know if that would be helpful enough!

**Comment by mwinckler** [comment]
@Alphy11 That sounds pretty neat. The same thoughts occurred to me regarding showing the initiative values - when playing with 2 players we find it manageable, but it does get tougher to keep track when there are 4 players. At minimum having some way to display the overall initiative values on-screen would be helpful; being able to display it on multiple devices would be fantastic.

We've talked about handling monster HP and status effects but nothing is actively underway yet. We did get [monster stats into JSON](https://github.com/mwinckler/GloomhavenMonsterStats/blob/master/monster_stats.json), which includes HP values in case that's useful to you. But I think our roadmap was tentatively to focus first on the monster stats themselves (getting them displayed appropriately on ability cards), then move on to HP/stat tracking later, so if you're working on HP tracking there'd be no overlap for at least some time to come. (And if you do pursue it - I'd love to keep up with your progress!)

**Comment by Alphy11** [comment]
Yeah, awesome to hear.  I am in the process of Getting a v1 together.  I'll share hopefully next week.



Wednesday, April 5, 2017, 1:01 PM -0700 from Matt Winckler <notifications@github.com>:

@Alphy11<https://github.com/Alphy11> That sounds pretty neat. The same thoughts occurred to me regarding showing the initiative values - when playing with 2 players we find it manageable, but it does get tougher to keep track when there are 4 players. At minimum having some way to display the overall initiative values on-screen would be helpful; being able to display it on multiple devices would be fantastic.

We've talked about handling monster HP and status effects but nothing is actively underway yet. We did get monster stats into JSON<https://github.com/mwinckler/GloomhavenMonsterStats/blob/master/monster_stats.json>, which includes HP values in case that's useful to you. But I think our roadmap was tentatively to focus first on the monster stats themselves (getting them displayed appropriately on ability cards), then move on to HP/stat tracking later, so if you're working on HP tracking there'd be no overlap for at least some time to come. (And if you do pursue it - I'd love to keep up with your progress!)

**Comment by Alphy11** [comment]
So, I got most of a prototype together if you guys would like to see what I'm doing.  It's still really ugly.  And the code is pretty ugly right now too, but I'll clean up some stuff soon.
Source:  https://github.com/Alphy11/gloomhaven-hp-tracker
Site: https://alphy11.github.io/#/  (currently doesn't do much but modify hp, I am working on mobile mostly.)
Mobile: https://alphy11.github.io/#/

**Comment by GinoGalotti** [comment]
It doesn't look that bad but.. is it really worth to have an extra app for that? My idea was including it on the gloomycompanion app, just adding a new menu on top (we already have the monster and scenario data), that way we can focus all the monster management work on the same app. We will also add a way to specify the monster level per deck... so it seems easier.  I was thinking about something really simple for the UI, like your work! (probably something smaller... maybe just an icon with the HP on top? and if you punch the icon, it reduces HP). 

If you prefer authoring something on your own, feel free of course... :) although we'd really appreciate having something like that as part of the companion!

**Comment by mwinckler** [comment]
@Alphy11 Slick! I like where you're going with this. Keep it up! :thumbsup:

**Comment by GinoGalotti** [comment]
https://ninjawithkillmoon.github.io/utilities/monsterHealth In case we want to get some inspiration!
